src/hooks/transactions/useSignTransactions.tsx

```diff
@@ -1,189 +1,236 @@
-                status: TransactionBatchStatusesEnum.signed,
-                transactions: Object.values(signedTransactions).map((tx) =>
-                  parseTransactionAfterSigning(tx)
-                )
-              })
-            );
-
-            if (
-              redirectAfterSign &&
-              !window.location.pathname.includes(redirectRoute)
-            ) {
-              window.location.href = redirectRoute;
-            }
-          }
-        } catch (err) {
-          const errorMessage =
-            (error as unknown as Error)?.message ||
-            (error as string) ||
-            'error signing transaction';
-
-          console.error('error signing transaction', errorMessage);
-          onCancel(errorMessage, sessionId);
-        }
+
+      const isProviderInitialized = await provider.init();
+
+      if (!isProviderInitialized) {
+        return;
       }
-    } catch (err) {
-      const errorMessage =
-        (error as unknown as Error)?.message ||
-        (error as string) ||
-        'error signing transaction';
-      console.error('error signing transaction', errorMessage);
+
+    } catch (error) {
+
+      const customErrorMessage = 'provider not intialized';
+      const errorMessage = (error as unknown as Error)?.message || (error as string) || customErrorMessage;
+      console.error(customErrorMessage, errorMessage);
       onCancel(errorMessage);
+      return;
+
+    };
+
+    try {
+
+      const signedTransactions = await provider.signTransactions(transactions);
+      const hasSameTransactions = Object.keys(signedTransactions).length === transactions.length;
+      const hasAllTransactionsSigned = signedTransactions && hasSameTransactions;
+      const shouldMoveTransactionsToSignedState = signedTransactions && hasAllTransactionsSigned;
+
+      if (!shouldMoveTransactionsToSignedState) {
+        return;
+      }
+
+      const signedTransactionsArray = Object.values(signedTransactions).map((tx) => parseTransactionAfterSigning(tx) );
+
+      dispatch(
+        moveTransactionsToSignedState({
+          sessionId,
+          transactions: signedTransactionsArray,
+          status: TransactionBatchStatusesEnum.signed
+        })
+      );
+
+      if (shouldRedirectAfterSign) {
+        window.location.href = redirectRoute;
+      }
+
+    } catch (err) {
+
+      const customErrorMessage = 'error signing transaction';
+      const errorMessage = (error as unknown as Error)?.message || (error as string) || customErrorMessage;
+      console.error(customErrorMessage, errorMessage);
+      onCancel(errorMessage, sessionId);
+    
+    };
+    
+  };
+
+  const signTransactions = async () => {
+
+    const hasSessionId = Boolean(transactionsToSign?.sessionId);
+    const { sessionId, transactions, callbackRoute } = transactionsToSign!;
+    const hasTransactions = Boolean(transactionsToSign?.transactions?.length);
+    const isSignEligible = transactionsToSign && hasSessionId && hasTransactions;
```

Comment:
hasSessionId and hasTransactions will be false if this is null, the check seems redundant